src/player.cpp

```diff
@@ -7552,7 +7551,7 @@ item::reload_option player::select_ammo( const item &base, std::vector<item::rel
     menu.callback = &cb;
 
     menu.query();
-    if( menu.ret < 0 || menu.ret >= ( int ) opts.size() ) {
+    if( menu.ret < 0 || static_cast<size_t>( menu.ret ) >= opts.size() ) {
```

Comment:
If you check the index via `size_t`, the check for `< 0` is redundant. If the number is less than zero, the unsigned form if it will be larger than the size of the container.

Reply:
> If the number is less than zero, the unsigned form if it will be larger than the size of the container.

I guess in theory, the size of the container can be as large as `static_cast<size_t>(-1)`, so that is not necessarily true? In any case, writing it that way may confuse other coders, so I think it's better to explicitly compare it to 0.